.github/workflows/main.yml

```diff
@@ -64,8 +64,31 @@ jobs:
         with:
           headers: "cache-control: no-cache"
           path: 'bin'
-          destination: 'packages.viam.com/apps/csi-camera/'
+          destination: 'packages.viam.com/apps/csi-camera/${{ env.TARGET }}'
           glob: '*aarch64*'
           parent: false
           gzip: false
-# TODO: Run cpp tests
+
+      - name: Get commit count
+          id: commit_count
+          run: echo "COMMIT_COUNT=$(git rev-list --count HEAD)" >> $GITHUB_EN
+
+      - name: Get latest release version
+        id: get_version
+        run: |
+          LATEST_RELEASE=$(curl -s https://api.github.com/repos/${{ github.repository }}/releases/latest | jq -r .tag_name)
+          echo "LATEST_RELEASE_VERSION=$LATEST_RELEASE" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Update module registry
+        uses: viamrobotics/upload-module@main
+        if: github.event_name == 'push'
+        with:
+          meta-path: ${{ env.META_PATH }}
+          module-path: module.tar.gz
+          platform: linux/arm64
+          version: ${{ env.LATEST_RELEASE_VERSION }}-${{ github.ref_name }}.${{ env.COMMIT_COUNT }}
```

Comment:
Adding latest main branch updates to registry here. 

Note: I am using `commit_count` instead of `run_number`